### ISSUE TITLE
Failed to build greptimedb for using centos based builder image

### ISSUE THREAD

**Issue by zyy17** [GreptimeTeam/greptimedb#3401]
### What type of bug is this?

Other

### What subsystems are affected?

Other

### Minimal reproduce step

The PR #3394 uses the option `--retry-connrefused` of `curl` to fetch dashboard assets, however, the current builder image `greptime/dev-builder-centos:latest` doesn't have the option because of using the old version `curl`:

```
docker run docker.io/greptime/dev-builder-centos:latest curl --retry-connrefused https://example.com
```

The building log: https://github.com/GreptimeTeam/greptimedb/actions/runs/8073087585/job/22056213664.

### What did you expect to see?

Build greptimedb of centos successfully.

### What did you see instead?

N/A

### What operating system did you use?

macOS 

### What version of GreptimeDB did you use?

latest

**Comment by zyy17** [comment]
cc @tisonkun

**Comment by zyy17** [comment]
IMO, Some solutions:

1. Upgrade `curl` for dev-builder-centos: I'm not sure whether it is an easy thing to upgrade the package for the legacy image;
2. Refactor the download script: If `--retry-connrefused` is necessary, maybe we can change another shell way to reach that. The main command can use the basic option of curl for the compatibility;

**Comment by tisonkun** [comment]
Yeah. This flag is available since curl 7.52.0. We can quickly workaround it by drop this flag. Or let me take a look if we can have a modern curl in the image.
